### DISCUSSION TITLE
Using Prisma for distributed system sharing the same database

### DISCUSSION THREAD

**Discussion by Jens-Rydholm-Humly** [prisma/prisma#24246] [Q&A]
### Question

We are using Prisma, and would like to use it for multiple applications sharing the same (PostgreSQL) database, with SQL foreign key constraints making sure their data doesn't become inconsistent with each other.

Simplified example:
* Application A has a Users table
* Application B has a Orders table, with a foreign key to the user placing the order
* Application B doesn't know (or need to know) anything else about the user except the ID (which it gets through e.g. a web service call)
* Each application has its own code repository, its own schema in the database, and its own `schema.prisma` file (describing only that application's data)
* We have some applications written in different frameworks (and languages), not using Prisma at all. We still want to constrain their related data with foreign keys

When we try to do this, Prisma wants to manage everything (including data it shouldn't need to know, owned by the other application). We get error messages like this one:

> The schema of the introspected database was inconsistent: The schema of the introspected database was inconsistent: Cross schema references are only allowed when the target schema is listed in the schemas property of your datasource. `b.orders` points to `a.users` in constraint `orders_user_id_fkey`. Please add `a` to your `schemas` property and run this command again.

Is there a way to make this work with Prisma? We really would prefer to have each application's concept of the data be minimal, being aware only of its own tables (and, maybe, the table it depends directly upon). Instead, adding this schema makes application B aware of all of application A's data, which (in a more complex scenario) would chain-react to cover the whole database.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment]
Hi @Jens-Rydholm-Humly 👋 

Thank you for sharing your usecase with us.

At the moment, it won't be possible to fully satisfy your usecase without including the other schema. This is necessary for the cross schema foreign key references between both applications.

You can create a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) and document your usecase so that our engineers can take a look.

**Reply by Jens-Rydholm-Humly** [reply]
Thank you for the clarification. For now, we are using a different tool (dbmate) for applying the migrations (to avoid Prisma introspection pulling everything else in), but we are still using Prisma for its ORM capabilities. This workaround unblocks the usecase for us, but we intend to file this as a feature request later (when we have the time to write it up properly).

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!

**Comment by zuffik** [comment]
Wow so no workaround for this? I just need to reference single table from one schema to another as a foreign key but prisma renders it impossible without pulling the entire other schema to my existing one.
